src/lexer.coffee

```diff
@@ -873,8 +858,12 @@ exports.Lexer = class Lexer
           # empty string.
           if i is 2 and firstEmptyStringIndex?
             @tokens.splice firstEmptyStringIndex, 2 # Remove empty string and the plus.
+          addTokenData token, initialChunk: yes if i is 0
+          addTokenData token, finalChunk: yes   if i is $
+          addTokenData token, {indent, quote, double}
+          addTokenData token, {heregex} if heregex
           token[0] = 'STRING'
-          token[1] = @makeDelimitedLiteral converted, options
+          token[1] = '"' + converted + '"'
```

Comment:
Tell me again why we're removing the quotes from the string value? Why not just pass them through and deal with them in the node class?

Reply:
@GeoffreyBooth ya this deserves further explanation

So previously we were always including quotes around `STRING` tokens since we were fully transforming/formatting them in the lexer before constructing the token. Specifically `makeDelimitedLiteral()` is responsible for adding the quotes. Then some would get output like that, but some (eg string chunks of a template literal) would then get their (artificialy generated) quotes re-stripped

So once I moved transformation/`makeDelimitedLiteral()` to the `nodes` side, it was sort of the default thing to do to pass them bare. But then I ran into the fact that in many places in the lexer, it just checks a string against `@value()`, so now string chunks that happened to match operators could cause false positives. So then I re-added quotes around `STRING` tokens to avoid this, but have the grammar strip them before passing to `new StringLiteral()` as I think it's weird to include fake quotes that weren't necessarily in the original source (eg if the `STRING` came from an interpolated string or a heregex) in the value that `StringLiteral` sees. You could have `StringLiteral::constructor()` strip them itself, but I'd rather contain it as a token artifact that the grammar is aware of (I also considered stripping them out in the token data cleanup pass at the end of the rewriter)

The state of `StringLiteral`s being quoted is still unresolved imo, eg `StringWithInterpolations::compileNode()` manually replaces the `value` of its `StringLiteral` children with the result of calling their `unquote()` method. But for now preserved the existing `@value` interface of `StringLiteral`s (which always includes quotes)